spec/models/binary_blob/purging_spec.rb

```diff
@@ -0,0 +1,41 @@
+RSpec.describe BinaryBlob do
+  let(:purge_time) { 1.month.ago.round }
+
+  context "::Purging" do
+    describe ".purge_by_scope" do
+      it "purges all non-resource rows and expired StateVarHash ones" do
+        # BinaryBlob with no resource
+        binary_blob1 = BinaryBlob.create
+
+        # BinaryBlob with a resource
+        report_result = FactoryBot.create(:miq_report_result)
+        binary_blob2  = BinaryBlob.create(:resource => report_result)
+
+        # StateVarHash blob that hasn't expired
+        binary_blob3 = BinaryBlob.create(:resource_type => "StateVarHash", :created_at => 1.week.ago.utc)
```

Comment:
In the latest version, resource_type won't be a "StateVarHash", so we should remove it from the tests too, so the records are more accurate.

```suggestion
        binary_blob3 = BinaryBlob.create(:created_at => 1.week.ago.utc)
```